cmd/lakefs/cmd/run.go

```diff
@@ -192,17 +179,41 @@ var runCmd = &cobra.Command{
 		// send metadata
 		bufferedCollector.CollectMetadata(metadata)
 
-		// wire actions
-		actionsService := actions.NewService(
-			ctx,
-			dbPool,
-			catalog.NewActionsSource(c),
-			catalog.NewActionsOutputWriter(c.BlockAdapter),
-			bufferedCollector,
-			cfg.GetActionsEnabled(),
-		)
-		c.SetHooksHandler(actionsService)
+		var multipartsTracker multiparts.Tracker
+		var actionsService actions.Service
+
+		if dbParams.KVEnabled {
+			kvStore, err := kv.Open(ctx, dbParams.Type, dbParams.ConnectionString)
+			if err != nil {
+				logger.WithError(err).Fatal("failed to open KV store")
+			}
+			defer kvStore.Close()
+			storeMessage := kv.StoreMessage{Store: kvStore}
+
+			multipartsTracker = multiparts.NewTracker(storeMessage)
+			actionsService = actions.NewKVService(
+				ctx,
+				storeMessage,
+				catalog.NewActionsSource(c),
+				catalog.NewActionsOutputWriter(c.BlockAdapter),
+				bufferedCollector,
+				cfg.GetActionsEnabled(),
+			)
+		} else {
+			multipartsTracker = multiparts.NewDBTracker(dbPool)
+			actionsService = actions.NewDBService(
+				ctx,
+				dbPool,
+				catalog.NewActionsSource(c),
+				catalog.NewActionsOutputWriter(c.BlockAdapter),
+				bufferedCollector,
+				cfg.GetActionsEnabled(),
+			)
+		}
```

Comment:
Just thinking out loud:
The following pattern repeats in various locations:
```
if dbParam.KVEnabled {
    OpenKV
    CreateSomeServiceWithKV
} else {
    CreateSomeServiceWithDB
}
```
It is (IMHO) hard to maintain, as it will keep growing with more modules `joining` club KV
Moreover, in case of changes - e.g. an addiional parameter is required to fix some bug etc - it will require multiple changes in multiple locations. There is, of course, the inevitable mega-modification of removing all DB support, once we are done implementing KV on all modules, and that, too, is going to be quite of an headache, considering the multiple locations that will require code change.
Long story short - maybe the `if dbParam.KVEnabled` decision should be moved to `actions.NewService` hiding  and call either `NewDBService`  or `NewSomethingElseService` and hiding the implementation details (Having a feature flag which controls the DB type) from the calling modules - `run` and `import` in that case, but can be more places for other modules.
This can aslo support possible future changes, similar to the move to KV, that may happen in the future.

Reply:
The "NewKVService" requires creating a KV store and ensuring it is closed when we are done with it. This will require to create and pass a KVstore **even if** it is not required for each service we want to create.
For now there are only 2 places we preform the flow. I suggest to defer it until we find this to be a real issue, also keep in mind this is a temporary code.